test/unit/middleware/missing.spec.ts

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file or at
+ * https://github.com/firebase/superstatic/blob/master/LICENSE
+ */
+
+import * as connect from "connect";
+import * as fs from "node:fs/promises";
+import * as request from "supertest";
+
+import * as fsProvider from "../../../src/providers/fs";
+import * as helpers from "../../helpers";
+import * as missingModule from "../../../src/middleware/missing";
+import * as Responder from "../../../src/responder";
+
+const missing = helpers.decorator(missingModule);
+
+describe("custom not found", () => {
+  const provider = fsProvider({
+    public: ".tmp",
+  });
+  let app: connect.Server;
+
+  beforeEach(async () => {
+    await fs.mkdir(".tmp");
```

Comment:
dumb question: wouldn't it be better to just use the sync versions of the fs functions in testing code?

Reply:
I don't know that it's _better_, but it would be an option. I think some of the `fs` library (not `fs/promises`) doesn't have sync versions, so waiting each of them is easy enough.